### PR TITLE
gh-122066: Test "roughly equivalent" implementations in itertools.rst

### PR DESCRIPTION
**Pull request by HarryLHW**
* Issue: gh-122066

----
📚 Documentation preview 📚: https://cpython-previews--122067.org.readthedocs.build/

### CODE REVIEW

**Comment by picnixz** [comment]
While I like the generic idea, I think it's better not to rely on a script extracting "roughly equivalent" paragraphs. It would be better to really test them explicitly, even though there are a lot of cases. 

Also, it could be preferrable to have doctests directly at the level of the itertools module with input/output then (I'll update my review tomorrow, now it's time to sleep 😴)

**Comment by skirpichev** [comment]
Instead of extracting code with flaky regexps, you could put these code snippets to separate files (which could be included with literalinclude sphinx directive).

**Comment by HarryLHW** [comment]
> you could put these code snippets to separate files (which could be included with literalinclude sphinx directive).
 Could you please give me some instructions or references for literalinclude sphinx directive?

File: Doc/library/itertools.rst [hunk 1]
```diff
@@ -218,10 +218,12 @@ loops that truncate the stream.
    Alternate constructor for :func:`chain`.  Gets chained inputs from a
    single iterable argument that is evaluated lazily.  Roughly equivalent to::
 
-      def from_iterable(iterables):
-          # chain.from_iterable(['ABC', 'DEF']) → A B C D E F
-          for iterable in iterables:
-              yield from iterable
+      class chain:
```
Comment (rhettinger) [inline Doc/library/itertools.rst:221]: Please don't edit the recipes. They are fine as-is.

**Comment by skirpichev** [comment]
> Could you please give me some instructions or references for literalinclude sphinx directive?

Sure, you can easily find them in the Doc/:
```
$ git grep literali.*py$ Doc/|head -3
Doc/library/datetime.rst:.. literalinclude:: ../includes/tzinfo_examples.py
Doc/library/difflib.rst:.. literalinclude:: ../includes/diff.py
Doc/library/difflib.rst:.. literalinclude:: ../includes/ndiff.py
```

Here is one case with many, you can see how it's rendered: https://docs.python.org/3.14/library/email.examples.html#email-examples

In fact, according to [docs](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-literalinclude), you can use just one single file for all recipes and select particular one with:
```rst
.. literalinclude:: ../include/itertools_equivalents.py
   :pyobject: accumulate
```

**Comment by HarryLHW** [comment]
> > Could you please give me some instructions or references for literalinclude sphinx directive?
> 
> Sure, you can easily find them in the Doc/:
> 
> ```
> $ git grep literali.*py$ Doc/|head -3
> Doc/library/datetime.rst:.. literalinclude:: ../includes/tzinfo_examples.py
> Doc/library/difflib.rst:.. literalinclude:: ../includes/diff.py
> Doc/library/difflib.rst:.. literalinclude:: ../includes/ndiff.py
> ```
> 
> Here is one case with many, you can see how it's rendered: https://docs.python.org/3.14/library/email.examples.html#email-examples
> 
> In fact, according to [docs](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-literalinclude), you can use just one single file for all recipes and select particular one with:
> 
> ```
> .. literalinclude:: ../include/itertools_equivalents.py
>    :pyobject: accumulate
> ```

Thank you!

**Pull request closed without merging**